src/IncludedCommands/Persistent/pbans.luau

```diff
@@ -51,86 +50,275 @@ local function GetCachedUsername(UserId: number): string
+            local Response = {}
+            local Individuals = {}
+            local Built = ""
+            for _, i in Error:split(",") do
+                if Built == "" then
+                    Built = i
+                else
+                    Built = `{Built},{i}`
                 end
-            end)
+                if Built:sub(Built:len()) == "}" then
+                    table.insert(Individuals,Built)
+                    Built = ""
+                end
+            end
+
+            for _, Message in Individuals do
+                local Split1 = Message:split(": {")
+                local ColonPosition,_ = Message:find(":")
+                if not ColonPosition then
+                    table.insert(Response,{
+                        UserId = -1,
+                        Lines = {
+                            "Unable to Parse Error Message:",
+                            Message
+                        }
+                    })
+                    continue
+                end
+                local RawJSON = Message:sub(ColonPosition + 1)
+                local ParsedJSON
+                pcall(function()
+                    ParsedJSON = HttpService:JSONDecode(RawJSON)
+                end)
+
+                --Isolate the UserId
+                local Split2 = Split1[1]:split(" ")
+                local UserId = Split2[#Split2]
+
+                if ParsedJSON then
+                    table.insert(Response,{
+                        UserId = tonumber(UserId),
+                        Lines = {
+                            `Code: {ParsedJSON.code}`,
+                            `Message:`,
+                            ParsedJSON.message,
+                            "",
+                            "Full Error Message:",
+                            Message
+                        }
+                    })
+                else
+                    table.insert(Response,{
+                        UserId = UserId,
+                        Lines = {
+                            `Unable to parse JSON`,
+                            "",
+                            "Full Error Message:",
+                            Message
+                        }
+                    })
+                end
+            end
+            return Response
         end
```

Comment:
This whole block and probably be simplified from my experience. It could be `HttpService::JSONDecode` on the error message and getting `message`, and then falling back to the original string.